skiko/src/awtMain/objectiveC/macos/DisplayLinkThrottler.mm

```diff
@@ -0,0 +1,162 @@
+#ifdef SK_METAL
+
+#import <jawt.h>
+#import <jawt_md.h>
+#import <QuartzCore/QuartzCore.h>
+#import <AppKit/AppKit.h>
+#import <stdatomic.h>
+
+@interface DisplayLinkThrottler : NSObject
+
+- (void)onVSync;
+
+@end
+
+static CVReturn displayLinkCallback(CVDisplayLinkRef displayLink, const CVTimeStamp *now, const CVTimeStamp *outputTime, CVOptionFlags flagsIn, CVOptionFlags *flagsOut, void *displayLinkContext) {
+    DisplayLinkThrottler *throttler = (__bridge DisplayLinkThrottler *)displayLinkContext;
+
+    [throttler onVSync];
```

Comment:
Will constant vsync affect battery on laptops?

Reply:
I doubt so, even on iOS it's overhead is next to non-existant.